### PR TITLE
[PKG-2306] statsmodels 0.14.0

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: https://www.statsmodels.org/devel/release/version0.14.0.html
License: https://github.com/statsmodels/statsmodels/blob/v0.14.0/LICENSE.txt
Requirements: 
- https://github.com/statsmodels/statsmodels/blob/v0.14.0/INSTALL.txt
- https://www.statsmodels.org/devel/install.html
- https://github.com/statsmodels/statsmodels/blob/v0.14.0/pyproject.toml
- https://github.com/statsmodels/statsmodels/blob/v0.14.0/requirements.txt
- https://github.com/statsmodels/statsmodels/blob/v0.14.0/setup.py

Actions:
1. Clean up the recipe:
- Remove skipping `py<37`
- Add flags `--no-build-isolation` to `script`
- Fix `numpy` in `host`
2. Use exact `setuptools_scm 7.0.4` in host, https://github.com/statsmodels/statsmodels/blob/6909efdd638390bd6079ca30d313b99344770180/pyproject.toml#L12
3. Update pinnings in `run` for `scipy` and `pandas`
4. Sort `host` and `run` dependencies
5. Remove `doc_source_url`

Notes:
- `plotnine 0.12.1` requires `statsmodels 0.14.0`, see https://github.com/has2k1/plotnine/blob/5ee0ed60b1341107ab80cc50d564721cbffda3a4/pyproject.toml#L32

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,66 +1,62 @@
...
     - {{ compiler('c') }}
   host:
     - python
+    - cython 0.29.32
+    - numpy {{ numpy }}
     - pip
-    # scipy needs 1.23 on python 3.11
-    - numpy 1.23         # [py==311]
-    - numpy {{ numpy }}  # [py!=311]
     - scipy
```
Comment (lorepirri) [inline recipe/meta.yaml:24]:
```suggestion
    - scipy 1.10.1
```

**Review by lorepirri — APPROVED** [review]
LGTM

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:24]: Fixed

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,66 +1,62 @@
...
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
   build:
     - {{ compiler('c') }}
   host:
     - python
+    - cython 0.29.32
```
Comment (cbouss) [inline recipe/meta.yaml:21]:
```suggestion
    - cython >=0.29.26,<3
```
There is no advantage in pinning cython to the lowest minor.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,66 +1,62 @@
...
     - pip
-    # scipy needs 1.23 on python 3.11
-    - numpy 1.23         # [py==311]
-    - numpy {{ numpy }}  # [py!=311]
-    - scipy
-    - cython >=0.29.32
+    - scipy 1.10.1
     - setuptools
-    - setuptools_scm >=7.0,<8
+    - setuptools_scm 7.0.4
```
Comment (cbouss) [inline recipe/meta.yaml:26]:
```suggestion
    - setuptools_scm >=7.0,<8
```
idem here

**Review by cbouss — APPROVED** [review]
The host pins are too strict to my liking, but otherwise it looks good to me.

**Pull request merged**